src/ducks/recurrence/search-with-existing-recurrence.spec.js

```diff
@@ -398,6 +398,70 @@ describe('recurrence scenario with multiple amounts and categories', () => {
     ])
   })
 
+  it('should not assign transaction to existing recurrence if not same account', () => {
```

Comment:
If you can optimize ;) see comment https://github.com/cozy/cozy-banks/pull/2235#discussion_r713925987

Reply:
ah oui ! J'ai fait le todo mais zappé celui-là, bien vu !

Reply:
🤔 après avoir commencé à le faire, je suis moins convaincu...

on perd le côté lecture linéaire du code je trouve. En plus ça rajoute autant de variables qu'il y a de transactions. Plus loin on a des cas qui gèrent plusieurs batch de transactions, comme Netflix et le Salaire par exemple, et pour des anciennes et nouvelles transactions, ça commence à faire beaucoup de variables et donc plus d'interrogation qu'autre chose.

Le sujet étant complexe, je me dis qu'une lecture simple et linéaire au moins facilite la chose.

Qu'en penses-tu ?

Reply:
Ça me paraissait lisible dans l'exemple et permettait de visualiser rapidement ce qui affecte les récurrences.
Après si tu considère qu'effectivement c'est moins lisible à l'usage alors d'accord.

Reply:
oui je pense qu'on peut valider en l'état

Reply:
ok je merge